app/rfid.py

```diff
@@ -11,21 +17,17 @@ def __init__(self):
             parity=serial.PARITY_NONE,
             stopbits=serial.STOPBITS_ONE,
             bytesize=serial.EIGHTBITS,
-            timeout=0,
+            timeout=10
         )
-        # ser.write(b'SD2\r\n')
+        # ser.write(b'SD2\r\n') # Sets transponder default to FDX-B(animal tag)
 
     def read(self):
-        # ser.reset_input_buffer()
+        ser.reset_input_buffer()
         ser.reset_output_buffer()
-        ser.write(b'rat\r\n')
-        a = ser.read(34).decode('utf-8')
-        if '?1' in a:
+        ser.write(b'sd2\r\n')
```

Comment:
When testing this I got 5% of the reads with a mixed up number.  Lets see with your method what we get.  I had to  do two reads and throw them away.  This gave us 0 false reads over the 9 days but was clumsy code.